optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -495,36 +383,156 @@ def _load_config(cls, config_name_or_path: Union[str, os.PathLike], **kwargs):
     def _save_config(self, save_directory):
         self.save_config(save_directory)
 
+    @property
+    def components(self) -> Dict[str, Any]:
+        components = {
+            "vae": self.vae,
+            "unet": self.unet,
+            "text_encoder": self.text_encoder,
+            "text_encoder_2": self.text_encoder_2,
+            "safety_checker": self.safety_checker,
+            "image_encoder": self.image_encoder,
+        }
```

Comment:
And for making this to work you might need to allow passing ORTModelXXX to `__init__()` instead of just bare ort session. Never mind, it's a small nit, we can put it as todo anyways!

Reply:
yes this one will need a bit more work for compatibility, I can do it in another PR.